File: PYTHON/1045 - Triangle Types.py
v = list(map(float,input().split()))

v.sort(reverse=True)

a,b,c = map(float,v)


if(a>=b+c):
    print("NAO FORMA TRIANGULO")
    
else:
    if((a*a) == (b*b)+(c*c)):
        print("TRIANGULO RETANGULO")
        
    if((a*a) > (b*b)+(c*c)):
        print("TRIANGULO OBTUSANGULO")
        
    if((a*a) < (b*b)+(c*c)):
        print("TRIANGULO ACUTANGULO")
        
    if(a==b and b == c):
        print("TRIANGULO EQUILATERO")
        
    if((a==b and c != a)or (b == c and a !=b)):
        print("TRIANGULO ISOSCELES")

File: C++/1045 - Triangle Types.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    float a,b,c,maior,meio,menor;

    cin>>a>>b>>c;
    maior = a;
    menor = a;


    if(b>maior)
        maior = b;

    if(c>maior)
        maior = c;
    
    if(b<menor)
        menor=b;

    if(c<menor)
        menor=c;

    meio = (a+b+c)-(maior+menor);

    a = maior;
    b = meio;
    c = menor;

    if(a>=b+c)
        cout<<"NAO FORMA TRIANGULO"<<endl;

    else{
        if(a*a == (b*b)+(c*c))
            cout<<"TRIANGULO RETANGULO"<<endl;
        if(a*a > (b*b)+(c*c))
            cout<<"TRIANGULO OBTUSANGULO"<<endl;
        if(a*a < (b*b)+(c*c))
            cout<<"TRIANGULO ACUTANGULO"<<endl;
 
        if(a == b && b==c)
            cout<<"TRIANGULO EQUILATERO"<<endl;

        if((a==b && a !=c)||(b == c && b!=a))
            cout<<"TRIANGULO ISOSCELES"<<endl;
    }
    

    return 0;
}